test/cancelable_operation_test.dart

```diff
@@ -387,6 +387,27 @@ void main() {
         expect(runThen().value, throwsA('error'));
         originalCompleter.operation.cancel();
       });
+
+      test('after completing with a future does not invoke `onValue`', () {
+        onValue = expectAsync1((_) => '', count: 0);
+        onCancel = null;
```

Comment:
Why not
```dart
 onCancel = expectAsync0(() => 'Fake');
```
It will be called (I assume), so might as well check that it is.

Reply:
We can test either of 2 possibilities. I suppose I could write them as separate test cases but I don't think it adds a lot of safety that we don't already get from other tests.

If `onCancel` is null then I can test that `operation.isCanceled` becomes true.
If `onCancel` is non-null then I can instead test that `operation.isCanceled` stays false, and it completes with the result of `onCancel`.